compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -677,6 +677,29 @@ object Trees {
     override def isType = expansion.isType
   }
 
+  /** A tree representing a quote `'{ expr }
+   *
+   *  @param  expr  The tree that was quoted
+   *  @param  tpt   The type of the tree that was quoted,
```

Comment:
How does this differ from expr.tpe?

Reply:
In the `staging` phase we use `tpt` to store the healed version of the type. This cannot be recomputed trivially from `expr.tpe`.

Reply:
Now staging uses` expr.tpe` to store the new version of type.